src/main/java/calculator/Calculator.java

```diff
@@ -0,0 +1,139 @@
+package calculator;
+
+import java.util.*;
+import java.lang.*;
+
+public class Calculator {
+    private static String operators = "+-*/";
+    private static String delimiters = "() " + operators;
+    private boolean flag = true;
+
+
+    public static int compute(String equation) {
+        Calculator n = new Calculator();
+        List<String> expression = n.parse(equation);
+        boolean flag = n.flag;
+        if (flag) {
+            return calc(expression);
+        }
+        return 0;
+    }
+
+
+    private static boolean isDelimiter(String token) {
+        if (token.length() != 1) return false;
+        for (int i = 0; i < delimiters.length(); i++) {
+            if (token.charAt(0) == delimiters.charAt(i)) return true;
+        }
+        return false;
+    }
+
+    private static boolean isOperator(String token) {
+        if (token.equals("u-")) return true;
+        for (int i = 0; i < operators.length(); i++) {
+            if (token.charAt(0) == operators.charAt(i)) return true;
+        }
+        return false;
+    }
+
+    private static boolean isFunction(String token) {
+        return token.equals("sqrt") || token.equals("cube") || token.equals("pow10");
+    }
+
+    private static int priority(String token) {
+        if (token.equals("(")) return 1;
+        if (token.equals("+") || token.equals("-")) return 2;
+        if (token.equals("*") || token.equals("/")) return 3;
+        return 4;
+    }
+
+    private List<String> parse(String infix) {
+        List<String> postfix = new ArrayList<>();
+        Deque<String> stack = new ArrayDeque<>();
+        StringTokenizer tokenizer = new StringTokenizer(infix, delimiters, true);
+        String prev = "";
+        String curr;
+        while (tokenizer.hasMoreTokens()) {
+            curr = tokenizer.nextToken();
+            if (!tokenizer.hasMoreTokens() && isOperator(curr)) {
+                flag = false;
+                return postfix;
+            }
+            if (curr.equals(" ")) continue;
+            if (isFunction(curr)) stack.push(curr);
+            else if (isDelimiter(curr)) {
+                if (curr.equals("(")) stack.push(curr);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/app/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4510206)